r/R/csv.R

```diff
@@ -699,15 +699,21 @@ write_csv_arrow <- function(x,
   if (is.null(write_options)) {
     write_options <- readr_to_csv_write_options(
       include_header = include_header,
-      batch_size = batch_size)
+      batch_size = batch_size
+    )
   }
 
   x_out <- x
+
+  if (inherits(x, "arrow_dplyr_query")) {
+    x <- as.data.frame(x)
+  }
+
   if (is.data.frame(x)) {
     x <- Table$create(x)
   }
 
-  assert_that(is_writable_table(x))
+  assert_that(is_writable_table(x) || inherits(x, c("Dataset", "arrow_dplyr_query")))
```

Comment:
Can we just deal with this in our `if` block below?

```suggestion
```

Reply:
Updated now